docker/bin/km-local

```diff
@@ -0,0 +1,174 @@
+    TIMEOUT=$(which gtimeout)
+fi
+
+PUBLIC_IP=$(publicip)
+
+set -e
+
+export K8SM_IP=${PUBLIC_IP}
+export K8SM_PORT=8888
+echo "Kubernetes: ${K8SM_IP}:${K8SM_PORT}"
+
+export MESOS_IP=${MESOS_IP:-$PUBLIC_IP}
+export MESOS_PORT=${MESOS_PORT:-5050}
+export MESOS_MASTER=${MESOS_IP}:${MESOS_PORT}
+echo "Mesos: ${MESOS_MASTER}"
+
+export ETCD_IP=${ETCD_IP:-$PUBLIC_IP}
+export ETCD_PORT=${ETCD_PORT:-4001}
+export ETCD_URL=http://${ETCD_IP}:${ETCD_PORT}
+echo "Etcd: ${ETCD_URL}"
+
+
+K8SM_CONFIG="$(pwd)/mesos-cloud.conf"
+echo "Config: ${K8SM_CONFIG}"
+if [ ! -f "$K8SM_CONFIG" ]; then
+    echo "Writing default config"
+    echo "[mesos-cloud]
+        http-client-timeout = 5s
+        state-cache-ttl     = 20s
+    " > $K8SM_CONFIG
+fi
+
+LOG_DIR=${LOG_DIR:-"/tmp/k8sm-logs"}
+mkdir -p ${LOG_DIR}
+echo "Log Dir: ${LOG_DIR}"
+
+echo "---------------------"
+
+
+SUBPROCS=()
+SUBPROC_NAMES=()
+
+pid_running () {
+    local CMD_PID=$1
+    return $(kill -0 ${CMD_PID} &> /dev/null)
+}
+
+# Add the pid of a sub-process or exit if it is not running
+add_subproc () {
+    local SUBPROC=$1
+    local SUBPROC_NAME="$2"
+    echo "Started ${SUBPROC_NAME} (${SUBPROC})"
+    if ! ps aux | grep -v grep | grep "${SUBPROC_NAME}" &> /dev/null; then
+        echo "Failed to start ${SUBPROC_NAME}"
+        exit 1
+    fi
+#    echo "Running ${SUBPROC_NAME} (${SUBPROC})"
+    SUBPROCS+=(${SUBPROC})
+    SUBPROC_NAME=${SUBPROC_NAME// /_} # replace spaces with underscores (bash arrays are space deliniated)
+    SUBPROC_NAMES+=(${SUBPROC_NAME})
+}
+
+# Kill subprocesses in reverse order of being started (to avoid log spam)
+kill_subprocs () {
+    SIGNAL=$1
+    echo "Shutting down [km-local] (Signal: ${SIGNAL}) ${#SUBPROCS[@]}"
+    if [ ! -z "${SUBPROCS}" ]; then
+        for ((i=${#SUBPROCS[@]}-1; i>=0; i--)); do
+            local SUBPROC=${SUBPROCS[i]}
+            local SUBPROC_NAME=${SUBPROC_NAMES[i]}
+            local SUBPROC_NAME=${SUBPROC_NAME//_/ } # replace underscores with spaces
+            if pid_running "${SUBPROC}"; then
+                echo "Killing ${SUBPROC_NAME} (${SUBPROC})"
+                #TODO(karl): propegate the recieved signal, instead of just TERM
+                kill ${SUBPROC} || true
+            fi
+        done
+    fi
+    if [ "${SIGNAL}" != "EXIT" ]; then
+        trap EXIT
+    fi
+}
+
+# propergate SIGINT & SIGTERM
+for sig in INT TERM EXIT; do
+    trap "kill_subprocs ${sig}" ${sig}
+done
+
+echo "Detecting etcd"
+await-service ${ETCD_IP} ${ETCD_PORT}
+echo "---------------------"
+
+
+echo "Detecting mesos-master"
+await-service ${MESOS_IP} ${MESOS_PORT}
+echo "---------------------"
+
+echo "Starting km apiserver &> ${LOG_DIR}/apiserver.log"
+km apiserver \
+  --address=${K8SM_IP} \
+  --etcd_servers=${ETCD_URL} \
+  --portal_net=10.10.10.0/24 \
+  --port=${K8SM_PORT} \
+  --cloud_provider=mesos \
+  --cloud_config=${K8SM_CONFIG} \
+  &> ${LOG_DIR}/apiserver.log \
```

Comment:
Log rotation might be necessary, depending on usage, but I suspect deleting after X time or over Y size is the more important part. We can definitely talk about this, but it's not quite critical yet. I don't expect to need log rotation (other than maybe truncation) during a test run.